File: .codex_review_extract/flutter-pos/lib/data/models/category.dart
class Category {
  final int idKategori;
  final String namaKategori;
  final String? deskripsi;
  final bool status;

  const Category({
    required this.idKategori,
    required this.namaKategori,
    this.deskripsi,
    required this.status,
  });

  factory Category.fromJson(Map<String, dynamic> json) => Category(
        idKategori: json['id_kategori'] as int,
        namaKategori: json['nama_kategori'] as String,
        deskripsi: json['deskripsi'] as String?,
        status: json['status'] == 1 || json['status'] == true,
      );

  Map<String, dynamic> toJson() => {
        'id_kategori': idKategori,
        'nama_kategori': namaKategori,
        'deskripsi': deskripsi,
        'status': status ? 1 : 0,
      };
}

File: .codex_review_extract/laravel-api/app/Models/Category.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Category extends Model
{
    protected $table = 'kategori_produk';
    protected $primaryKey = 'id_kategori';
    public $timestamps = true;

    protected $fillable = [
        'nama_kategori',
        'deskripsi',
        'status',
    ];

    protected $casts = [
        'status' => 'boolean',
    ];

    public function products(): HasMany
    {
        return $this->hasMany(Product::class, 'id_kategori', 'id_kategori');
    }
}
